src/QsCompiler/QirGeneration/Subtransformations/StatementTransformation.cs

```diff
@@ -33,10 +33,10 @@ public QirStatementTransformation(GenerationContext sharedState, TransformationO
 
         public override QsStatement OnStatement(QsStatement stm)
         {
-            QsNullable<QsLocation> loc = stm.Location;
-            this.SharedState.DIManager.LocationStack.Push(loc);
+            this.SharedState.DIManager.StatementLocationStack.Push(stm.Location.Item);
+            this.SharedState.DIManager.EmitLocation(Position.Zero);
```

Comment:
The Position.Zero was supposed to be the offset relative to the current namespace and statement stack. I renamed the functions for clarity. Having two statements on the same line isn't an issue, there just are two emitted locations with the same line. Ultimately it only matters if there is at least one emitted location with whatever line you want to set the breakpoint on. I am emitting column information as well, so technically this would allow you to set two breakpoints on the same line if you were debugging with column specificity.

Reply:
The Position.Zero was supposed to be the offset relative to the current namespace and statement stack. I renamed the functions for clarity. Having two statements on the same line isn't an issue, there just are two emitted locations with the same line. Ultimately it only matters if there is at least one emitted location with whatever line you want to set the breakpoint on. I am emitting column information as well, so technically this would allow you to set two breakpoints on the same line if you were debugging with column specificity.